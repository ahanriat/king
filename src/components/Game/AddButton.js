import ReactNative from 'react-native';
import React from 'react';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  content: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  text: {
    fontSize: 30,
    color: 'white',
  },
});

class AddButton extends Component {

  render() {
    return (
      <View style={this.props.style}>
        <View style={styles.container}>
          <TouchableOpacity
            onPress={this.props.onPress}
          >
            <View style={styles.content}>
              <Text style={styles.text}>{this.props.text}</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

}

AddButton.propTypes = {
  onPress: PropTypes.func,
  text: PropTypes.string,
  style: PropTypes.number,
};

module.exports = AddButton;
