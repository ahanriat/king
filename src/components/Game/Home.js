import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  Text,
} = ReactNative;

const {
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    color: '#ffffff',
  },

});

class GameHome extends Component {

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Ok</Text>
      </View>
    );
  }

}

GameHome.propTypes = {
};

export default GameHome;
