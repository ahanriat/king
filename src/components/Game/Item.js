import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Animated,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-start',
    paddingLeft: 30,
    marginTop: 15,
    marginBottom: 15,
    height: 40,
  },
  titleWrapper: {
    flex: 1,

  },
  title: {
    fontFamily: 'Montserrat-Light',
    fontSize: 19,
    color: '#ffffff',
  },

  subTitleWrapper: {
    flex: 1,
    flexDirection: 'row',
  },
  winner: {
    fontFamily: 'Montserrat-Light',
    color: '#ffffff',
  },
  looser: {
    fontFamily: 'Montserrat-Light',
    color: '#ffffff',
  },
});
const ACTION_TIMER = 400;

class GameList extends Component {

  constructor() {
    super();

    this.state = {
      pressAction: new Animated.Value(0),
    };
  }

  componentWillMount() {
    this.progressValue = 0;
    this.state.pressAction.addListener((v) => { this.progressValue = v.value; });
  }

  handlePressIn() {
    Animated.timing(this.state.pressAction, {
      duration: ACTION_TIMER,
      toValue: 1,
    }).start(() => this.animationActionComplete());
  }

  handlePressOut() {
    if (this.progressValue < 1) {
      this.props.onSelect(this.props.game.getId());
    }
    Animated.timing(this.state.pressAction, {
      duration: 0,
      toValue: 0,
    }).start();
  }

  animationActionComplete() {
    if (this.progressValue === 1) {
      this.props.onLongPress(this.props.game.getId());
    }
  }

  renderPlayers() {
    const { game } = this.props;

    if (game.getWinnerId()) {
      return (
        <View style={styles.subTitleWrapper}>
          <View style={styles.winnerWrappr}>
            <Text style={styles.winner}>{game.getWinnerId()}</Text>
          </View>
          <View style={styles.looserWrapper}>
            <Text style={styles.looser}>{game.getLooserId()}</Text>
          </View>
        </View>
      );
    }
    return false;
  }

  render() {
    const { game } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity
          style={styles.closeButton}
          onPressIn={() => this.handlePressIn()}
          onPressOut={() => this.handlePressOut()}
        >
          <View style={styles.titleWrapper}>
            <Text style={styles.title}>{game.getTitle()}</Text>
          </View>

          {this.renderPlayers()}
        </TouchableOpacity>
      </View>
    );
  }

}

//

GameList.propTypes = {
  game: PropTypes.object,
  onSelect: PropTypes.func,
  onLongPress: PropTypes.func,
};

module.exports = GameList;
