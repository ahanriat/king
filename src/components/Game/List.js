import React from 'react';
import ReactNative from 'react-native';

import GameItem from './Item';

const {
  StyleSheet,
  View,
  ScrollView,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {

  },

});

class GameList extends Component {
  render() {
    const { games, onItemSelected, onItemLongPress } = this.props;
    const sortedGames = games.sortBy((value) => {
      return parseInt(value.get('id'), 10);
    }).reverse();
    return (
      <View style={this.props.style}>
        <ScrollView style={styles.container}>
          {sortedGames.valueSeq().map((game) => {
            return (
              <GameItem
                onLongPress={onItemLongPress}
                onSelect={onItemSelected}
                key={game.getId()}
                game={game}
              />
            );
          })}
        </ScrollView>
      </View>

    );
  }

}

GameList.propTypes = {
  style: PropTypes.number,
  games: PropTypes.object,
  onItemSelected: PropTypes.func,
  onItemLongPress: PropTypes.func,
};

module.exports = GameList;
