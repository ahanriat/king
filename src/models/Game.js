import { Record, List } from 'immutable';

const GameRecord = Record({
  id: null,
  title: null,
  playerIds: new List(),
  winnerId: null,
  looserId: null,
  createdAt: new Date(),
  closedAt: null,
});

const pad = (num, size) => {
  let s = `${num}`;
  while (s.length < size) s = `0${s}`;
  return s;
};

class Game extends GameRecord {
  getId() {
    return this.get('id');
  }

  getTitle() {
    return `Partie #${pad(this.get('id'), 3)}`;
  }

  getPlayerIds() {
    return this.get('playerIds');
  }

  getWinnerId() {
    return this.get('winnerId');
  }

  getLooserId() {
    return this.get('looserId');
  }

  getCreatedAt() {
    return this.get('createdAt');
  }

  getClosedAt() {
    return this.get('closedAt');
  }
}

module.exports = Game;
