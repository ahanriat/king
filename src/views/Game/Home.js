import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Immutable, { Map, List } from 'immutable';
import React from 'react';
import ReactNative from 'react-native';
import { Actions } from 'react-native-router-flux';

import gameActions from '../../actions/gameActions';
import scoreActions from '../../actions/scoreActions';

import ScoreList from '../../components/Score/List';
import Header from '../../components/Header';

import CloseIcon from '../../assets/images/ic_close.png';
import AddIcon from '../../assets/images/ic_add.png';
import DashedLine from '../../assets/images/dashed_line.png';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image,
} = ReactNative;

const {
  Component,
  PropTypes,
} = React;


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
  },

  content: {
    flex: 8,
    flexDirection: 'row',
  },
  colTotal: {
    flex: 4,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    flexDirection: 'column',
    paddingTop: 40,
  },
  colAdd: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  colScore: {
    flex: 9,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    paddingTop: 40,
  },

  addButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  addButtonIconWrapper: {
    flex: 1,
  },
  addButtonIcon: {
    width: 20,
    height: 20,
  },
  addButtonDashWrapper: {
    flex: 10,
  },
  addButtonDash: {
    flex: 1,
    width: 10,
    flexDirection: 'column',
  },

  playerScoreWrapper: {
    marginLeft: 20,
    marginBottom: 30,
    height: 50,
  },
  playerScore: {
    fontFamily: 'Montserrat-Black',
    fontSize: 30,
    color: '#ffffff',
  },
  playerName: {
    fontFamily: 'Montserrat-Regular',
    fontSize: 12,
    color: '#ffffff',
  },

  backgroundImage: {
    flex: 1,
    resizeMode: 'stretch',
    marginTop: 10,
  },
});

class Feed extends Component {

  constructor(props) {
    super(props);

    const game = props.game.get('list').get(props.gameId);

    this.state = {
      game,
      scores: new Map(),
      rounds: new Map(),
      players: new List(),
      currentRound: 1,
    };
  }

  componentDidMount() {
    this.props.scoreActions.fetch(this.props.gameId);
    this.props.gameActions.fetch(this.props.gameId);
  }

  componentWillReceiveProps(nextProps) {
    const scoreList = nextProps.score.get('list');

    const scores = scoreList.filter((score) => {
      return score.getGameId() === this.props.gameId;
    });
    const rounds = Immutable.Map(scores.reduce((result, item) => {
      const newresult = result;
      const scoreRound = newresult[item.getRound()] || { id: item.getRound(), scores: [] };
      scoreRound.scores.push(item);
      newresult[item.getRound()] = scoreRound;
      return newresult;
    }, {}));
    const lastRound = parseInt(rounds.keySeq().map((key) => parseInt(key, 10)).max(), 10) || 0;
    const game = nextProps.game.get('list').get(nextProps.gameId);
    const players = game.getPlayerIds();
    this.setState({
      currentRound: lastRound + 1,
      game,
      rounds,
      scores,
      players,
    });
  }

  getPlayerScore(id) {
    const total = this.state.scores.reduce((result, item) => {
      let newResult = result;
      if (id === item.getPlayerId()) {
        newResult += item.getValue();
      }
      return newResult;
    }, 0);

    return total;
  }

  handleAddScore() {
    Actions.gameScore({ gameId: this.props.gameId, round: this.state.currentRound });
  }

  handleClose() {
    Actions.pop();
  }

  handleEditScore(round) {
    Actions.gameScore({ gameId: this.props.gameId, round: round.id, scores: round.scores });
  }

  render() {
    return (
      <View style={styles.container}>
        <Header
          title={this.state.game.getTitle()}
          onLeftButtonPress={(() => this.handleClose())}
          buttonLeftImage={CloseIcon}
        />
        <View style={styles.content}>
          <View style={styles.colTotal}>

            {this.state.players.map((player) => {
              return (
                <View key={`total_${player}`} style={styles.playerScoreWrapper}>
                  <Text
                    style={styles.playerScore}
                    numberOfLines={1}
                  >
                    {this.getPlayerScore(player)}
                  </Text>
                  <Text style={styles.playerName}>{player}</Text>
                </View>
              );
            })}
          </View>
          <View style={styles.colAdd}>
            <TouchableOpacity
              style={styles.addButton}
              onPress={(() => this.handleAddScore())}
            >
              <View style={styles.addButtonIconWrapper}>
                <Image
                  style={styles.addButtonIcon}
                  source={AddIcon}
                />
              </View>
              <View style={styles.addButtonDashWrapper} />
            </TouchableOpacity>
            <Image
              style={styles.backgroundImage}
              source={DashedLine}
            />
          </View>
          <View style={styles.colScore}>
            <ScoreList
              players={this.state.players}
              rounds={this.state.rounds}
              handleRoundLongPress={(round) => this.handleEditScore(round)}
            />
          </View>
        </View>
      </View>
    );
  }

}

Feed.propTypes = {
  gameId: PropTypes.string.isRequired,
  score: PropTypes.object.isRequired,
  game: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  gameActions: PropTypes.object.isRequired,
  scoreActions: PropTypes.object.isRequired,
  routes: PropTypes.object,
};

const mapStateToProps = (state) => ({
  score: state.score,
  game: state.game,
  player: state.player,
  routes: state.routes,
});
const mapDispatchToProps = (dispatch) => ({
  gameActions: bindActionCreators(gameActions, dispatch),
  scoreActions: bindActionCreators(scoreActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Feed);
