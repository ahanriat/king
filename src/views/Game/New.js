import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import React from 'react';
import ReactNative from 'react-native';
import { Actions } from 'react-native-router-flux';

import ParticipantChoice from '../../components/Participant/Choice';
import gameActions from '../../actions/gameActions';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
} = ReactNative;

const {
  Component,
  PropTypes,
} = React;


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2DB0CD',
  },
  finishButton: {
    flex: 1,
  },
});

class Feed extends Component {

  constructor() {
    super();
    this.state = {
      selectedPlayers: [],
    };
  }

  componentDidMount() {
  }

  handlePlayersChange(players) {
    this.setState({
      selectedPlayers: players,
    });
  }

  handleNext() {
    const game = {
      title: '',
      playerIds: this.state.selectedPlayers,
    };
    this.props.gameActions.create(game);
  }

  render() {
    const { player } = this.props;
    return (
      <View style={styles.container}>
        <ParticipantChoice
          players={player.get('list')}
          onChange={((ids) => this.handlePlayersChange(ids))}
        />
        <TouchableOpacity
          style={styles.finishButton}
          onPress={(() => this.handleNext())}
        >
          <Text>Next</Text>
        </TouchableOpacity>
      </View>
    );
  }

}

Feed.propTypes = {
  game: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  gameActions: PropTypes.object.isRequired,
  routes: PropTypes.object,
};

const mapStateToProps = (state) => ({
  game: state.game,
  player: state.player,
  routes: state.routes,
});
const mapDispatchToProps = (dispatch) => ({
  gameActions: bindActionCreators(gameActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Feed);
