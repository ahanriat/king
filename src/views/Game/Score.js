import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import React from 'react';
import ReactNative from 'react-native';
import { Actions } from 'react-native-router-flux';
import isNumber from 'is-number';

import gameActions from '../../actions/gameActions';
import scoreActions from '../../actions/scoreActions';

import NextIcon from '../../assets/images/ic_next.png';
import CloseIcon from '../../assets/images/ic_close.png';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  TextInput,
  Image,
} = ReactNative;

const {
  Component,
  PropTypes,
} = React;


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000000',
  },

  scoreContainer: {
    flex: 10,
    alignItems: 'center',
    justifyContent: 'center',

  },
  scoreContent: {
    marginTop: -40,
  },
  score: {
    fontFamily: 'Montserrat-Black',
    fontSize: 150,
    color: '#FFFFFF',
    textAlign: 'center',
  },
  player: {
    fontFamily: 'Montserrat-Regular',
    fontSize: 12,
    color: '#FFFFFF',
    textAlign: 'center',
  },

  input: {

  },

  header: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  closeButton: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  closeButtonIcon: {
    width: 20,
    height: 20,
  },
  headerTitle: {
    flex: 5,
    fontFamily: 'Montserrat-Black',
    fontSize: 16,
    justifyContent: 'center',
    textAlign: 'center',
    color: '#FFFFFF',
  },

  headerAfter: {
    flex: 1,
  },

  addButton: {
    flex: 1,
    backgroundColor: '#F8E71C',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  addButtonBefore: {
    flex: 1,
  },
  addButtonText: {
    flex: 8,
    fontFamily: 'Montserrat-Black',
    fontSize: 18,
    justifyContent: 'center',
    textAlign: 'center',
  },
  addButtonIconWrapper: {
    flex: 1,
  },
  addButtonIcon: {
    width: 20,
    height: 20,
  },
});

class GameScore extends Component {

  constructor(props) {
    super(props);

    const game = props.game.get('list').get(props.gameId);
    const players = game.getPlayerIds();

    const playerScores = players.map((player) => {
      let value = 0;
      let score = null;
      if (this.props.scores) {
        const filteredCores = this.props.scores.filter((item) => {
          return item.getPlayerId() === player;
        });
        score = filteredCores[0];
        value = score.getValue() || 0;
      }
      return {
        name: player,
        score: value,
        defaultScore: score,
      };
    });
    const currentPlayer = playerScores.first();

    this.state = {
      players: playerScores,
      game,
      currentPlayer,
      currentIndex: 0,
      buttonText: 'JOUEUR SUIVANT',
      title: game.getTitle(),
    };
  }

  handleOnChange(text) {
    let intValue = parseInt(text, 10);
    if (!text) {
      intValue = 0;
    }
    if (isNumber(intValue)) {
      const currentPlayer = this.state.currentPlayer;
      currentPlayer.score = intValue;
      this.setState({
        currentPlayer,
      });
    }
  }

  handleAddScore() {
    let id = new Date().getTime().toString();
    if (this.state.currentPlayer.defaultScore) {
      id = this.state.currentPlayer.defaultScore.getId();
    }
    const scoreObject = {
      playerId: this.state.currentPlayer.name,
      value: this.state.currentPlayer.score,
      round: this.props.round,
      gameId: this.props.gameId,
      id,
    };
    this.props.scoreActions.create(scoreObject);
    const nextIndex = this.state.currentIndex + 1;
    if (nextIndex >= this.state.players.count()) {
      Actions.pop();
    } else {
      let buttonText = this.state.buttonText;
      if (nextIndex === this.state.players.count() - 1) {
        buttonText = 'TERMINÃ‰';
      }
      this.setState({
        buttonText,
        currentIndex: nextIndex,
        currentPlayer: this.state.players.get(nextIndex),
      });
    }
  }

  handleClose() {
    Actions.pop();
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <TouchableOpacity
            style={styles.closeButton}
            onPress={(() => this.handleClose())}
          >
            <Image
              style={styles.closeButtonIcon}
              source={CloseIcon}
            />
          </TouchableOpacity>
          <Text style={styles.headerTitle}>{this.state.title}</Text>
          <View style={styles.headerAfter} />
        </View>
        <TextInput
          style={styles.input}
          onChangeText={(text) => this.handleOnChange(text)}
          value={this.state.currentPlayer.score.toString()}
          autoFocus
        />
        <View style={styles.scoreContainer}>
          <View style={styles.scoreContent}>
            <Text style={styles.score}>{this.state.currentPlayer.score}</Text>
            <Text style={styles.player}>{this.state.currentPlayer.name}</Text>
          </View>
        </View>
        <TouchableOpacity
          style={styles.addButton}
          onPress={(() => this.handleAddScore())}
        >
          <View style={styles.addButtonBefore} />
          <Text style={styles.addButtonText}>{this.state.buttonText}</Text>
          <View style={styles.addButtonIconWrapper} >
            <Image
              style={styles.addButtonIcon}
              source={NextIcon}
            />
          </View>
        </TouchableOpacity>
      </View>
    );
  }

}

GameScore.propTypes = {
  scores: PropTypes.array,
  round: PropTypes.number.isRequired,
  gameId: PropTypes.string.isRequired,
  game: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  gameActions: PropTypes.object.isRequired,
  scoreActions: PropTypes.object.isRequired,
  routes: PropTypes.object,
};

const mapStateToProps = (state) => ({
  score: state.score,
  game: state.game,
  player: state.player,
  routes: state.routes,
});
const mapDispatchToProps = (dispatch) => ({
  gameActions: bindActionCreators(gameActions, dispatch),
  scoreActions: bindActionCreators(scoreActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(GameScore);
