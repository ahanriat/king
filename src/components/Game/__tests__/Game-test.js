import React from 'react';
import { Map } from 'immutable';
import renderer from 'react-test-renderer'; // eslint-disable-line

import Game from '../../../models/Game';

import GameAddButton from '../AddButton';
import GameList from '../List';
import GameItem from '../Item';

describe('Game', () => {
  it('render Button', () => {
    const props = {
      text: 'Test String',
      onPress: () => {},
      selected: false,
    };
    const tree = renderer.create(
      <GameAddButton {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render List', () => {
    const data = {
      id: 'id1',
      title: 'title game',
      playerIds: ['id1', 'id2'],
      createdAt: new Date(1987, 11, 12).toLocaleString('en-US', { timeZone: 'America/New_York' }),
      closedAt: new Date(1987, 11, 12).toLocaleString('en-US', { timeZone: 'America/New_York' }),
      winnerId: 'id1',
    };
    const game = new Game(data);
    let gameList = new Map();
    gameList = gameList.set('id1', game);
    const props = {
      games: gameList,
    };
    const tree = renderer.create(
      <GameList {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render Item', () => {
    const data = {
      id: 'id1',
      title: 'title game',
      playerIds: ['id1', 'id2'],
      createdAt: new Date(1987, 11, 12),
      closedAt: new Date(1995, 11, 12),
      winnerId: 'id1',
    };
    const game = new Game(data);

    const props = {
      game,
    };
    const tree = renderer.create(
      <GameItem {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
