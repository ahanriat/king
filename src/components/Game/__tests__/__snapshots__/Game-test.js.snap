exports[`Game render Button 1`] = `
<View
  style={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center"
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "opacity": 1
        }
      }
      testID={undefined}>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center"
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 30
            }
          }>
          Test String
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Game render Item 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "justifyContent": "center",
      "paddingLeft": 30
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "stretch",
        "height": 70,
        "justifyContent": "center",
        "opacity": 1
      }
    }
    testID={undefined}>
    <View
      style={Object {}}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "flex": 1,
            "fontFamily": "Montserrat-Light",
            "fontSize": 19,
            "marginBottom": 5
          }
        }>
        Partie #id1
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start"
        }
      }>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "marginRight": 13
          }
        }>
        <Image
          source={1}
          style={
            Array [
              undefined,
              undefined
            ]
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#F8E71C"
              },
              Object {
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 12,
                "marginLeft": 4
              }
            ]
          }>
          id1
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "marginRight": 13
          }
        }>
        <Image
          source={1}
          style={
            Array [
              undefined,
              undefined
            ]
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#9B9B9B"
              },
              Object {
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 12,
                "marginLeft": 4
              }
            ]
          } />
      </View>
    </View>
  </View>
</View>
`;

exports[`Game render List 1`] = `
<View
  style={undefined}>
  <ScrollView
    style={Object {}}>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 30
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function bound touchableHandleResponderGrant]}
        onResponderMove={[Function bound touchableHandleResponderMove]}
        onResponderRelease={[Function bound touchableHandleResponderRelease]}
        onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
        onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
        onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
        style={
          Object {
            "alignItems": "stretch",
            "height": 70,
            "justifyContent": "center",
            "opacity": 1
          }
        }
        testID={undefined}>
        <View
          style={Object {}}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "flex": 1,
                "fontFamily": "Montserrat-Light",
                "fontSize": 19,
                "marginBottom": 5
              }
            }>
            Partie #id1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start"
            }
          }>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginRight": 13
              }
            }>
            <Image
              source={1}
              style={
                Array [
                  undefined,
                  undefined
                ]
              } />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#F8E71C"
                  },
                  Object {
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 12,
                    "marginLeft": 4
                  }
                ]
              }>
              id1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginRight": 13
              }
            }>
            <Image
              source={1}
              style={
                Array [
                  undefined,
                  undefined
                ]
              } />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#9B9B9B"
                  },
                  Object {
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 12,
                    "marginLeft": 4
                  }
                ]
              } />
          </View>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
